feat(community)_: add version to image url to let clients update

jrainville opened the pull request:
Fixes https://github.com/status-im/status-desktop/issues/16688

Since we use the local image server to show the community image, the URL never changes when we update the image, since it's served using a query string containing the community ID. eg: `https://Localhost:46739/communityDescriptionImages?communityID=0x03c5ece7da362d31199fb02d632f85fdf853af57d89c3204b4d1e90c6ec13bb23c&name=thumbnail` Because of that, the clients cannot know if the image was updated, so they had to force update the image every time, which was inefficient.

We discovered this issue when I refactored the community client code in Desktop so that we only update the changed properties of a community instead of reseting the whole thing.

The solution I came up with in the PR is to add a `version` to the URL when we detect that the image changed. This let,s the clients detect when the image was updated without having to do any extra logic.

jrainville committed:
feat(community)_: add version to image url to let clients update

Fixes https://github.com/status-im/status-desktop/issues/16688

Since we use the local image server to show the community image, the URL never changes when we update the image, since it's served using a query string containing the community ID. eg: `https://Localhost:46739/communityDescriptionImages?communityID=0x03c5ece7da362d31199fb02d632f85fdf853af57d89c3204b4d1e90c6ec13bb23c&name=thumbnail`
Because of that, the clients cannot know if the image was updated, so they had to force update the image every time, which was inefficient.

We discovered this issue when I refactored the community client code in Desktop so that we only update the changed properties of a community instead of reseting the whole thing.

The solution I came up with in the PR is to add a `version` to the URL when we detect that the image changed. This let's the clients detect when the image was updated without having to do any extra logic.

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (80)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | dfc4dc80 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6118/1/) | 2024-11-22 20:06:26 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241122-200148-dfc4dc-pr6118.zip) |
| :heavy_check_mark: | dfc4dc80 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6118/1/) | 2024-11-22 20:06:49 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241122-200145-dfc4dc-pr6118.zip) |
| :heavy_multiplication_x: | dfc4dc80 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/1/) | 2024-11-22 20:06:50 | ~4 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/1//consoleText) |
| :heavy_check_mark: | dfc4dc80 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6118/1/) | 2024-11-22 20:06:56 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241122-200146-dfc4dc-pr6118.zip) |
| :heavy_check_mark: | dfc4dc80 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6118/1/) | 2024-11-22 20:07:52 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241122-200144-dfc4dc-pr6118.aar) |
| :heavy_check_mark: | dfc4dc80 | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6118/1/) | 2024-11-22 20:08:00 | ~6 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241122-200153-dfc4dc-pr6118.zip) |
| :heavy_check_mark: | dfc4dc80 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6118/1/) | 2024-11-22 20:09:07 | ~7 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241122-200150-dfc4dc-pr6118.zip) |
| :heavy_multiplication_x: | dfc4dc80 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/1/) | 2024-11-22 20:35:24 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/1//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 65d153d2 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/2/) | 2024-11-26 19:29:43 | ~3 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/2//consoleText) |
| :heavy_check_mark: | 65d153d2 | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6118/2/) | 2024-11-26 19:30:46 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241126-192617-65d153-pr6118.zip) |
| :heavy_check_mark: | 65d153d2 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6118/2/) | 2024-11-26 19:30:57 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241126-192551-65d153-pr6118.zip) |
| :heavy_check_mark: | 65d153d2 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6118/2/) | 2024-11-26 19:31:01 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241126-192553-65d153-pr6118.zip) |
| :heavy_check_mark: | 65d153d2 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6118/2/) | 2024-11-26 19:31:07 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241126-192550-65d153-pr6118.zip) |
| :heavy_check_mark: | 65d153d2 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6118/2/) | 2024-11-26 19:31:51 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241126-192549-65d153-pr6118.aar) |
| :heavy_check_mark: | 65d153d2 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6118/2/) | 2024-11-26 19:33:19 | ~7 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241126-192554-65d153-pr6118.zip) |
| :heavy_multiplication_x: | 65d153d2 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/3/) | 2024-11-26 19:33:32 | ~3 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/3//consoleText) |
| :heavy_check_mark: | 65d153d2 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6118/3/) | 2024-11-26 19:35:49 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241126-193102-65d153-pr6118.zip) |
| :heavy_check_mark: | 65d153d2 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6118/3/) | 2024-11-26 19:36:13 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241126-193106-65d153-pr6118.zip) |
| :heavy_check_mark: | 65d153d2 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6118/3/) | 2024-11-26 19:36:42 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241126-193112-65d153-pr6118.zip) |
| :heavy_multiplication_x: | 65d153d2 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/2/) | 2024-11-26 19:37:37 | ~11 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/2//consoleText) |
| :heavy_check_mark: | 65d153d2 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6118/3/) | 2024-11-26 19:37:58 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241126-193156-65d153-pr6118.aar) |
| :heavy_check_mark: | 65d153d2 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6118/3/) | 2024-11-26 19:40:50 | ~7 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241126-193325-65d153-pr6118.zip) |
| :heavy_multiplication_x: | 65d153d2 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/3/) | 2024-11-26 19:47:38 | ~9 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/3//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 64d4e662 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/4/) | 2024-11-26 21:34:01 | ~3 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/4//consoleText) |
| :heavy_multiplication_x: | 64d4e662 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/4/) | 2024-11-26 21:34:20 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/4//consoleText) |
| :heavy_check_mark: | 64d4e662 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6118/3/) | 2024-11-26 21:35:14 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241126-213021-64d4e6-pr6118.zip) |
| :heavy_check_mark: | 64d4e662 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6118/4/) | 2024-11-26 21:35:23 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241126-213016-64d4e6-pr6118.zip) |
| :heavy_check_mark: | 64d4e662 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6118/4/) | 2024-11-26 21:35:26 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241126-213017-64d4e6-pr6118.zip) |
| :heavy_check_mark: | 64d4e662 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6118/4/) | 2024-11-26 21:35:34 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241126-213013-64d4e6-pr6118.zip) |
| :heavy_check_mark: | 64d4e662 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6118/4/) | 2024-11-26 21:36:23 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241126-213012-64d4e6-pr6118.aar) |
| :heavy_check_mark: | 64d4e662 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6118/4/) | 2024-11-26 21:38:53 | ~8 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241126-213017-64d4e6-pr6118.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 16a9843b | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/5/) | 2024-11-26 21:37:52 | ~3 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/5//consoleText) |
| :heavy_check_mark: | 16a9843b | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6118/4/) | 2024-11-26 21:39:19 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241126-213517-16a984-pr6118.zip) |
| :heavy_check_mark: | 16a9843b | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6118/5/) | 2024-11-26 21:40:31 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241126-213530-16a984-pr6118.zip) |
| :heavy_check_mark: | 16a9843b | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6118/5/) | 2024-11-26 21:40:40 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241126-213529-16a984-pr6118.zip) |
| :heavy_check_mark: | 16a9843b | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6118/5/) | 2024-11-26 21:41:03 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241126-213538-16a984-pr6118.zip) |
| :heavy_check_mark: | 16a9843b | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6118/5/) | 2024-11-26 21:42:42 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241126-213629-16a984-pr6118.aar) |
| :heavy_multiplication_x: | 16a9843b | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/5/) | 2024-11-26 21:44:48 | ~10 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/5//consoleText) |
| :heavy_check_mark: | 16a9843b | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6118/5/) | 2024-11-26 21:46:34 | ~7 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241126-213902-16a984-pr6118.zip) |
| | | | | | | |
| :heavy_multiplication_x: | b8d2544b | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/6/) | 2024-11-27 15:10:19 | ~4 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/6//consoleText) |
| :heavy_check_mark: | b8d2544b | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6118/5/) | 2024-11-27 15:10:28 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241127-150611-b8d254-pr6118.zip) |
| :heavy_check_mark: | b8d2544b | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6118/6/) | 2024-11-27 15:11:52 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241127-150606-b8d254-pr6118.zip) |
| :heavy_check_mark: | b8d2544b | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6118/6/) | 2024-11-27 15:12:13 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241127-150602-b8d254-pr6118.aar) |
| :heavy_check_mark: | b8d2544b | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6118/6/) | 2024-11-27 15:12:53 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241127-150605-b8d254-pr6118.zip) |
| :heavy_check_mark: | b8d2544b | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6118/6/) | 2024-11-27 15:12:58 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241127-150607-b8d254-pr6118.zip) |
| :heavy_check_mark: | b8d2544b | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6118/6/) | 2024-11-27 15:13:37 | ~7 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241127-150607-b8d254-pr6118.zip) |
| :heavy_multiplication_x: | b8d2544b | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/6/) | 2024-11-27 15:16:52 | ~10 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | b3659a16 | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6118/6/) | 2024-11-27 15:58:18 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241127-155401-b3659a-pr6118.zip) |
| :heavy_check_mark: | b3659a16 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/7/) | 2024-11-27 15:59:37 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/7//consoleText) |
| :heavy_check_mark: | b3659a16 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6118/7/) | 2024-11-27 15:59:48 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241127-155355-b3659a-pr6118.zip) |
| :heavy_check_mark: | b3659a16 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6118/7/) | 2024-11-27 15:59:49 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241127-155353-b3659a-pr6118.zip) |
| :heavy_check_mark: | b3659a16 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6118/7/) | 2024-11-27 15:59:58 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241127-155352-b3659a-pr6118.zip) |
| :heavy_check_mark: | b3659a16 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6118/7/) | 2024-11-27 16:00:18 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241127-155352-b3659a-pr6118.aar) |
| :heavy_check_mark: | b3659a16 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6118/7/) | 2024-11-27 16:02:16 | ~8 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241127-155357-b3659a-pr6118.zip) |
| :heavy_multiplication_x: | b3659a16 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/7/) | 2024-11-27 16:23:18 | ~29 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 47f41c96 | [#7](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6118/7/) | 2024-11-28 19:54:27 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241128-194951-47f41c-pr6118.zip) |
| :heavy_check_mark: | 47f41c96 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/8/) | 2024-11-28 19:54:39 | ~4 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/8//consoleText) |
| :heavy_check_mark: | 47f41c96 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6118/8/) | 2024-11-28 19:55:11 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241128-194946-47f41c-pr6118.zip) |
| :heavy_check_mark: | 47f41c96 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6118/8/) | 2024-11-28 19:55:36 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241128-194945-47f41c-pr6118.zip) |
| :heavy_check_mark: | 47f41c96 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6118/8/) | 2024-11-28 19:55:51 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241128-194943-47f41c-pr6118.aar) |
| :heavy_check_mark: | 47f41c96 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6118/8/) | 2024-11-28 19:56:07 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241128-194944-47f41c-pr6118.zip) |
| :heavy_check_mark: | 47f41c96 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6118/8/) | 2024-11-28 19:57:19 | ~7 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241128-194947-47f41c-pr6118.zip) |
| :heavy_multiplication_x: | 47f41c96 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/8/) | 2024-11-28 20:19:36 | ~29 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/8//consoleText) |
| :heavy_check_mark: | 47f41c96 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/9/) | 2024-11-29 16:02:18 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/9//consoleText) |
| :heavy_check_mark: | 47f41c96 | [#8](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6118/8/) | 2024-11-29 18:59:24 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241129-185434-47f41c-pr6118.zip) |
| :heavy_multiplication_x: | 47f41c96 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/9/) | 2024-11-29 18:59:41 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/9//consoleText) |
| :heavy_check_mark: | 47f41c96 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6118/9/) | 2024-11-29 19:00:07 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241129-185428-47f41c-pr6118.zip) |
| :heavy_check_mark: | 47f41c96 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6118/9/) | 2024-11-29 19:00:51 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241129-185426-47f41c-pr6118.aar) |
| :heavy_check_mark: | 47f41c96 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6118/9/) | 2024-11-29 19:00:59 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241129-185427-47f41c-pr6118.zip) |
| :heavy_check_mark: | 47f41c96 | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6118/9/) | 2024-11-29 19:02:01 | ~7 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241129-185430-47f41c-pr6118.zip) |
| :heavy_check_mark: | 47f41c96 | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6118/9/) | 2024-11-29 19:02:25 | ~7 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241129-185430-47f41c-pr6118.zip) |
| :heavy_multiplication_x: | 47f41c96 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/10/) | 2024-11-29 19:25:08 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/10//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | a3008747 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/10/) | 2024-11-29 19:04:24 | ~4 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/10//consoleText) |
| :heavy_check_mark: | a3008747 | [#9](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6118/9/) | 2024-11-29 19:04:54 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241129-185931-a30087-pr6118.zip) |
| :heavy_check_mark: | a3008747 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6118/10/) | 2024-11-29 19:05:32 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241129-190011-a30087-pr6118.zip) |
| :heavy_check_mark: | a3008747 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6118/10/) | 2024-11-29 19:06:31 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241129-190056-a30087-pr6118.aar) |
| :heavy_check_mark: | a3008747 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6118/10/) | 2024-11-29 19:06:54 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241129-190109-a30087-pr6118.zip) |
| :heavy_check_mark: | a3008747 | [#10](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6118/10/) | 2024-11-29 19:07:46 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241129-190229-a30087-pr6118.zip) |
| :heavy_check_mark: | a3008747 | [#10](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6118/10/) | 2024-11-29 19:09:40 | ~7 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241129-190212-a30087-pr6118.zip) |
| :heavy_check_mark: | a3008747 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/11/) | 2024-11-29 19:54:52 | ~29 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/11//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 9c7fd7fa | [#11](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/11/) | 2024-12-02 16:58:12 | ~4 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/11//consoleText) |
| :heavy_check_mark: | 9c7fd7fa | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6118/11/) | 2024-12-02 16:58:51 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241202-165320-9c7fd7-pr6118.zip) |
| :heavy_check_mark: | 9c7fd7fa | [#10](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6118/10/) | 2024-12-02 16:58:53 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241202-165328-9c7fd7-pr6118.zip) |
| :heavy_check_mark: | 9c7fd7fa | [#11](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6118/11/) | 2024-12-02 16:59:04 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241202-165321-9c7fd7-pr6118.zip) |
| :heavy_check_mark: | 9c7fd7fa | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6118/11/) | 2024-12-02 16:59:07 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241202-165319-9c7fd7-pr6118.zip) |
| :heavy_check_mark: | 9c7fd7fa | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6118/11/) | 2024-12-02 16:59:50 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241202-165317-9c7fd7-pr6118.aar) |
| :heavy_check_mark: | 9c7fd7fa | [#11](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6118/11/) | 2024-12-02 17:06:07 | ~12 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241202-165323-9c7fd7-pr6118.zip) |
| :heavy_multiplication_x: | 9c7fd7fa | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/12/) | 2024-12-02 17:23:45 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/12//consoleText) |
| :heavy_check_mark: | 9c7fd7fa | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/13/) | 2024-12-03 15:30:10 | ~28 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/13//consoleText) |
| | | | | | | |
| :heavy_check_mark: | dd62d1c9 | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6118/12/) | 2024-12-03 18:20:49 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241203-181528-dd62d1-pr6118.zip) |
| :heavy_check_mark: | dd62d1c9 | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6118/12/) | 2024-12-03 18:21:01 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241203-181525-dd62d1-pr6118.aar) |
| :heavy_check_mark: | dd62d1c9 | [#12](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/12/) | 2024-12-03 18:21:19 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6118/12//consoleText) |
| :heavy_check_mark: | dd62d1c9 | [#11](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6118/11/) | 2024-12-03 18:21:21 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241203-181536-dd62d1-pr6118.zip) |
| :heavy_check_mark: | dd62d1c9 | [#12](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6118/12/) | 2024-12-03 18:21:31 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241203-181529-dd62d1-pr6118.zip) |
| :heavy_check_mark: | dd62d1c9 | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6118/12/) | 2024-12-03 18:21:36 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241203-181527-dd62d1-pr6118.zip) |
| :heavy_check_mark: | dd62d1c9 | [#12](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6118/12/) | 2024-12-03 18:23:08 | ~7 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241203-181531-dd62d1-pr6118.zip) |
| :heavy_check_mark: | dd62d1c9 | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/14/) | 2024-12-03 18:45:47 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6118/14//consoleText) |

ilmotta commented on protocol/communities/community.go:
```diff
@@ -1211,6 +1212,19 @@ func (o *Community) MemberIdentity() *ecdsa.PublicKey {
+	for imageType, newImage := range description.Identity.Images {
+		oldImage, ok := o.config.CommunityDescription.Identity.Images[imageType]
+		if ok {
+			// FIXME this does not work!!!!!
```
Maybe a leftover comment?

caybro commented on server/server_media.go:
```diff
@@ -211,12 +211,13 @@ func (s *MediaServer) MakeCommunityTokenImagesURL(communityID string, chainID ui
 	u.RawQuery = url.Values{
 		"communityID": {communityID},
 		"name":        {name},
+		"version":     {strconv.FormatUint(uint64(version), 10)},
```
`uint64` here and `uint32` everywhere else?

jrainville replied:
`FormatUint` accepts uint64 only :shrug:

caybro replied:
:roll_eyes: go doesn't have some sort of generics or method overloads? :open_mouth:

caybro reviewed (approved):
LGTM

igor-sirotin commented on protocol/protobuf/chat_identity.proto:
```diff
@@ -58,6 +58,10 @@ message IdentityImage {
 
+  // version iterator of the image since images use a localUrl to be shown,
+  // this iterator makes sure the image URL changes when a new image is given
+  uint32 version = 6;
```
Wait, so `version` is not just local, it becomes part of `CommunityDescription`? 🤔 
Sounds a bit weird to be like that, like this problem should be solved solely on client side, not affecting network messages

jrainville replied:
Yeah I agree it should be local only, but I didn't find an easy way to do so, since locally, we use the `IdentityImage` protobuf in the image server.

I'm very open for suggestions to make it locally, but apart from creating some sort of new column or table, I don't know how to make it work.

In the end, this solution is pretty unobtrusive anyway IMO.

igor-sirotin replied:
@jrainville I think I'm not convinced, modifying protocol for local needs sounds counter-intuitive to me.

We should probably wrap the `IdentityImage` when passing it to `ServerMedia`, and add the `Version` property to the wrapper. Dunno, or make some method of the media server to update the version of specific image.

> _creating some sort of new column or table_

We don't need to persist the version, it's enough to have it only during a single run. After restart, version can go again from `1`, should work, right?

jrainville replied:
> We don't need to persist the version, it's enough to have it only during a single run. After restart, version can go again from 1, should work, right?

That's true. I'll try to see if I can do as you say

osmaczko replied:
I agree with Igor here.

At first glance, the best would be to use/extend [CommunityChanges](https://github.com/status-im/status-go/blob/ad28f15531c76c216f08118d4055a140ee52f80f/protocol/communities/community_changes.go#L20) with image change recognition and then populate it to client.

jrainville replied:
> We don't need to persist the version, it's enough to have it only during a single run. After restart, version can go again from 1, should work, right?

Ok, I tried something simple and I realized that we indeed need to save it somewhere. At least a local cache. However, we do not have a local cache for communities. The closest we have is the `encrypted_community_description_cache` which is a table. It is also what the image server uses to serve the images.

Just to make sure the problem is clear, I can't just pass the updated version to the image server, since the Edit happens in one thread/timeframe and the image serving is a completely different one, so the image server knows nothing about image updates.
It serves using the community cache table, which is populated from the decrypted community description. So we need the version or at least an indicator that the images were updated in there.

So either I create a new local cache for image versions per community, or I add a new column `encrypted_community_description_cache` to save the version, or leave it as I did it. Or you guys have a better option?

osmaczko commented on protocol/communities/manager.go:
```diff
@@ -1650,6 +1650,11 @@ func (m *Manager) EditCommunity(request *requests.EditCommunity) (*Community, er
 	if community.IsControlNode() {
 		community.increaseClock()
+		// Update the community cache. This makes sure that the image server has access to the new image if needed
+		_, _, err = m.preprocessDescription(request.CommunityID, community.config.CommunityDescription)
```
This doesn't seem quite right. First, the community is edited, followed by the generation of a new description. Subsequently, the new description undergoes reprocessing, which involves the complete mechanism of decryption and populating encryption-specific cache tables. In this case, the cache is used to store _unencrypted_ descriptions for subsequent processing. However, the invariant is broken in this scenario, and I’m not sure how this ties into the image server.

jrainville replied:
It took me a while to understand what was going on. The image server relies on the community cache to serve the images, ie `encrypted_community_description_cache`.

When calling `EditCommunity` the image does get saved in the `communities` DB. However, the cache does **not** get updated straight away.
From what i could gather, it gets updated the next time `getByID` gets called, which is just a race condition at this point.

Calling `preprocessDescription` after editing the clock makes sure the new description gets saved in the cache DB.

Maybe I could call `SaveDecryptedCommunityDescription` directly instead of `preprocessDescription`? It would be clearer in the code, but `preprocessDescription` does a bit more handling with regards to members, so I thought it would be safer to use that instead.

@osmaczko do you have a suggestion on how to make it better? Maybe a better comment?

osmaczko replied:
I see. I don't understand why the image server relies on `encrypted_community_description_cache`. The purpose of this cache is tightly coupled with the hash ratchet, meaning the cache is invalidated whenever we receive a key that was previously missing for that community. Therefore, in theory, it wouldn't work correctly if all keys for a given community are received.

This was introduced in the following PR: https://github.com/status-im/status-go/pull/5336. @qfrank, do you recall why the `encrypted_community_description_cache` table was used there? The media server should ideally just use the `GetByID` method (must be injected somehow).

@jrainville would you mind fixing that within the scope of this PR please?

jrainville replied:
> @jrainville would you mind fixing that within the scope of this PR please?

I don't mind. I'll just wait for @qfrank 's reply to make sure there isn't something we don,t know about. Maybe the cache serves faster?

qfrank replied:
> to make sure there isn't something we don,t know about. Maybe the cache serves faster?

It was easier to implement fetching image through db as media server already contains db instance, that's all :)

osmaczko replied:
I understand the reasoning. However, `encrypted_community_description_cache` should not be used to retrieve the community description, as it can be invalidated at any time. This means the entry might be removed and become inaccessible. Moreover, `GetByID` should be used instead because it applies events on top of the raw community description. With the current implementation, if an admin changes the image (logo or banner) and the change is not yet approved by the control node, then the image will not be read properly by the media server.

jrainville replied:
Fixed here: https://github.com/status-im/status-go/pull/6127

I'll rebase this PR on top of that other one

osmaczko reviewed (changes requested):
Sorry, I need to request changes for this one.

jrainville commented:
@igor-sirotin @osmaczko I rebased this branch on top of https://github.com/status-im/status-go/pull/6127 as I refactored the media sever to not use the community cache.
So I don't need `preprocessDescription` anymore :raised_hands: 

As for the version being in the protobuf, I sadly didn't find a good solution. I'm still hoping you guys can think of something :confused:

jrainville commented:
Patryk gave me a good suggestion:
> or another trick to do in manager.go:
mediaServer.ImageVersion(func(communityID string) { return imageVersions[communiyID]})

> And in UpdateImageVersions you update the imageVersions map instead ChatIdentity 
This way clients would work fine without changes 

Using a local map in the manager

jrainville commented:
I updated the PR to go with Patryk's recommendation. I no longer need the protobuf changes and the version isn't saved anymore. Please let me know what you think :smile:

osmaczko commented on protocol/communities/manager.go:
```diff
@@ -479,6 +481,13 @@ func NewManager(
 
 	if mediaServer != nil && !reflect.ValueOf(mediaServer).IsNil() {
+		mediaServer.SetCommunityImageVersionReader(func(communityID string) uint32 {
+			version, ok := manager.communityImageVersions[communityID]
+			if !ok {
+				return 0
+			}
+			return version
```
```suggestion
			return manager.communityImageVersions[communityID]
```

osmaczko commented on protocol/communities/manager.go:
```diff
@@ -1616,6 +1625,15 @@ func (m *Manager) UpdatePubsubTopicPrivateKey(topic string, privKey *ecdsa.Priva
 	return m.transport.RemovePubsubTopicKey(topic)
 }
 
+func (m *Manager) incrementCommunityImageVersion(communityID string) {
```
Would be great to add a comment explaining why this is needed.

osmaczko commented on server/server_media.go:
```diff
@@ -243,6 +255,7 @@ func (s *MediaServer) MakeCommunityImageURL(communityID, name string) string {
 	u.RawQuery = url.Values{
 		"communityID": {communityID},
 		"name":        {name},
+		"version":     {strconv.FormatUint(uint64(s.getCommunityImageVersion(communityID)), 10)},
```
```suggestion
		"version":     {fmt.Sprintf("%d", s.getCommunityImageVersion(communityID))},
```

osmaczko commented on protocol/communities/manager.go:
```diff
@@ -1616,6 +1625,15 @@ func (m *Manager) UpdatePubsubTopicPrivateKey(topic string, privKey *ecdsa.Priva
 }
 
+func (m *Manager) incrementCommunityImageVersion(communityID string) {
+	version, ok := m.communityImageVersions[communityID]
+	if !ok {
+		// First time the image is updated, so update the version to 1 (it was 0 by default)
+		m.communityImageVersions[communityID] = 1
+	}
+	m.communityImageVersions[communityID] = version + 1
```
```suggestion
	m.communityImageVersions[communityID] = m.communityImageVersions[communityID] + 1
```

jrainville commented:
FYI @igor-sirotin @osmaczko these two tests seem flaky. They failed on my PR by work locally:
- TestRetrieveBigCommunity
- TestService_IncrementalUpdateMixed

I'll restart them on my PR

igor-sirotin commented:
> * TestRetrieveBigCommunity
> * TestService_IncrementalUpdateMixed

@jrainville yep they're flaky 💯 
There're some more, appeared all at once 💀

osmaczko commented:
> FYI @igor-sirotin @osmaczko these two tests seem flaky. They failed on my PR by work locally:
> 
> * TestRetrieveBigCommunity
> * TestService_IncrementalUpdateMixed
> 
> I'll restart them on my PR

You don't need to restart unless one of the tests fails three times in a row, which is highly unlikely even for flaky ones.

igor-sirotin commented:
> FYI @igor-sirotin @osmaczko these two tests seem flaky. They failed on 
> You don't need to restart unless one of the tests fails three times in a row, which is highly unlikely even for flaky ones.

btw This only works if a test fails. But if it panics, then there's no retries 😐

jrainville commented:
@igor-sirotin @qfrank @ilmotta can I get one more review please. I updated a lot of the old review wasn't up to date

ilmotta commented on protocol/communities/manager.go:
```diff
@@ -1621,6 +1626,15 @@ func (m *Manager) UpdatePubsubTopicPrivateKey(topic string, privKey *ecdsa.Priva
 
+// Managing the version of community images is necessary because image URLs are "constant"
+// For eg: https://Localhost:46739/communityDescriptionImages?communityID=[ID]&name=thumbnail
+// So the clients have way of knowing that they need to reload the image
```
Missing `no` word? `So the clients have no way of knowing [...]`

qfrank reviewed (approved):
LGTM

BTW, the solution seems pretty specific for community image, I'm wondering would it be better to implement it in a common way for media server, because this issue could also exist with other kind of image, e.g. profile image  🤔

caybro commented:
> LGTM
> 
> BTW, the solution seems pretty specific for community image, I'm wondering would it be better to implement it in a common way for media server, because this issue could also exist with other kind of image, e.g. profile image 🤔

Yeah, we have exactly the same problem with profile image as well

jrainville commented:
> LGTM
> 
> BTW, the solution seems pretty specific for community image, I'm wondering would it be better to implement it in a common way for media server, because this issue could also exist with other kind of image, e.g. profile image 🤔

Yeah, when I do the same fix for the profile images, I'll do a pass to generalize the functions, if possible.

But currently on Desktop, only the community image **showed** the issue, because we don't have the "hack" of adding a timestamp to the link each time it changes.

I'll work on that later in the next milestone if I have time
